### ISSUE TITLE
Creation of binding causing operator pod to crash

### ISSUE THREAD

**Issue by hollisc** [IBM/cloud-operators#215]
**Issue:** Creation of a `binding.ibmcloud` for an existing `databases-for-postgresql` services.ibmcloud for failed and crashed the operator pod
**Expected Result:** Binding resource created and a secret with service credentials
**Actual Result:** Binding resource not created and operator pod crashed with following logs
**Steps to reproduce:**
1. Log in to IBM Cloud and set resource group, org and space.
- `ibmcloud login`
- `ibmcloud target -g <resource group> -o <org> -s <space>`
2. Log in to OCP 4.4 cluster.
3. Install IBM Cloud Operator
- `curl -sL https://raw.githubusercontent.com/IBM/cloud-operators/master/hack/configure-operator.sh | bash -s -- install`
4. Create management namespace. 
5. Create secret and configmap for use by the operator in the management namespace.  The name of the secret and configmap contains ref to the application namespace.
6. Create an application namespace.
7. Provision instance of postgresql (refer to yaml used in comments).
- `oc apply -f <service.ibmcloud yaml>`
8. Create binding for postgresql service (refer to yaml used in comments).
- `oc apply -f <binding.ibmcloud yaml>`
9. Operator pod crashes. 

**Logs:**
```
2020-09-22T19:39:07.335Z	INFO	controllers.Binding	Context	{"binding": "hc-dev/posgresql-binding", "is": {"org":"devops-gse","space":"dev","region":"us-south","resourcegroupid":"d256d988fbf64884b6d25b1dfae2706b","resourcelocation":"us-south","user":"hollisc@ca.ibm.com"}}
2020-09-22T19:39:08.367Z	INFO	controllers.Binding	Creating	{"credentials": "posgresql-binding"}
E0922 19:39:08.368105       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 465 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x14146e0, 0x2227060)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/runtime/runtime.go:48 +0x82
panic(0x14146e0, 0x2227060)
	/usr/local/go/src/runtime/panic.go:969 +0x166
github.com/ibm/cloud-operators/controllers.(*BindingReconciler).getResourceServiceCredentials(0xc0002d8000, 0xc000624710, 0xc00000a960, 0xc008fbfb90, 0xc008fbfb90, 0x0, 0x0, 0xc00000a960, 0xc00969d160)
	/workspace/controllers/binding_controller.go:463 +0x6a
github.com/ibm/cloud-operators/controllers.(*BindingReconciler).createCredentials(0xc0002d8000, 0x17cb360, 0xc000040040, 0xc000624710, 0xc00000a960, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/workspace/controllers/binding_controller.go:459 +0x170
github.com/ibm/cloud-operators/controllers.(*BindingReconciler).Reconcile(0xc0002d8000, 0xc0012de25a, 0x6, 0xc00061af20, 0x11, 0x0, 0xbfd2b296bb45b8e8, 0xc0000f0090, 0xc00044a368)
	/workspace/controllers/binding_controller.go:260 +0x12f8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001e2000, 0x14641a0, 0xc001212d00, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:256 +0x161
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001e2000, 0xc000255400)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:232 +0xae
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0001e2000)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0004788b0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:152 +0x5f
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0004788b0, 0x3b9aca00, 0x0, 0x165a501, 0xc00015c240)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0004788b0, 0x3b9aca00, 0xc00015c240)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:193 +0x305
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12acf4a]

goroutine 465 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/runtime/runtime.go:55 +0x105
panic(0x14146e0, 0x2227060)
	/usr/local/go/src/runtime/panic.go:969 +0x166
github.com/ibm/cloud-operators/controllers.(*BindingReconciler).getResourceServiceCredentials(0xc0002d8000, 0xc000624710, 0xc00000a960, 0xc008fbfb90, 0xc008fbfb90, 0x0, 0x0, 0xc00000a960, 0xc00969d160)
	/workspace/controllers/binding_controller.go:463 +0x6a
github.com/ibm/cloud-operators/controllers.(*BindingReconciler).createCredentials(0xc0002d8000, 0x17cb360, 0xc000040040, 0xc000624710, 0xc00000a960, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/workspace/controllers/binding_controller.go:459 +0x170
github.com/ibm/cloud-operators/controllers.(*BindingReconciler).Reconcile(0xc0002d8000, 0xc0012de25a, 0x6, 0xc00061af20, 0x11, 0x0, 0xbfd2b296bb45b8e8, 0xc0000f0090, 0xc00044a368)
	/workspace/controllers/binding_controller.go:260 +0x12f8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0001e2000, 0x14641a0, 0xc001212d00, 0x0)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:256 +0x161
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0001e2000, 0xc000255400)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:232 +0xae
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0001e2000)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:211 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0004788b0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:152 +0x5f
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0004788b0, 0x3b9aca00, 0x0, 0x165a501, 0xc00015c240)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc0004788b0, 0x3b9aca00, 0xc00015c240)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.2/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.5.0/pkg/internal/controller/controller.go:193 +0x305
```

**Comment by hollisc** [comment]
My env is OCP 4.4 cluster and I set up a management namespace that contains the secret and configmap.  Below are the yamls I used. 

`service.ibmcloud` yaml :
```
apiVersion: ibmcloud.ibm.com/v1alpha1
kind: Service
metadata:
    name: postgresql-dev
spec:
    plan: standard
    serviceClass: databases-for-postgresql
```

`binding.ibmcloud` yaml: 
```
apiVersion: ibmcloud.ibm.com/v1
kind: Binding
metadata:
    name: posgresql-binding
spec:
    serviceName: postgresql-dev

```

**Comment by JohnStarich** [comment]
If I'm understanding the problem correctly, this should be fixed in v0.3.3+. Does the new version address that issue for you?

**Comment by hollisc** [comment]
Was using `cloudoperators/ibmcloud-operator:0.3.1`.  I upgraded to v0.3.2 and was able to successfully create a binding.

This issue can be closed.
